### DISCUSSION TITLE
Feel free to ask anything! 🙋

### DISCUSSION THREAD

**Discussion by dxsup** [CloudDetail/apo#4] [Announcements]
Feel free to ask anything! 🙋

**Comment by litaixun** [comment]
kindling 项目还在维护吗，apo 和 kindling 有什么区别，kindling 如何监控 containerd 或 docker 等数据？

**Reply by dxsup** [reply]
kindling项目已经转交其他团队维护，我不再维护该项目。APO是一个完整的可观测性平台，开箱即用可以采集Metrics、Traces、Logs等，另外充分利用TraceProfiling和北极星指标用于分析问题。
kindling天然支持监控容器环境数据。

**Reply by litaixun** [reply]
收到，感谢作者百忙之中抽空回复，由于最近 kindling 项目的提问无人回复，想多问一点，kindling 项目是否有推荐的学习资料文档链接，想深入研究并把 kindling 用在生产上收集一些内核事件以解决一些业务问题，但感觉目前能找到的资料不太够，直接看源码感觉有点茫然。

**Reply by dxsup** [reply]
kindling 的公开资料都在官方网站上了，网站的[仓库地址](https://github.com/KindlingProject/website).
除此之外 kindling 的 eBPF 底层代码基于这个[开源项目](https://github.com/falcosecurity/libs)，同时这个底层也用于开源项目 [falco](https://github.com/falcosecurity/falco) 和 [sysdig](https://github.com/draios/sysdig)，所以感兴趣的话相关资料都可以看看。

**Reply by litaixun** [reply]
大佬，看 kindling 源码时，有一个疑惑，就是各种内核事件是如何与页面上展示出来的每个请求关联起来的呀，是在 eBPF 层处理好的吗，主要是哪些字段来关联的呢，是如何提取这些信息然后关联在一起的呀？目前 kindling 社区暂时没人回复，如果您有时间，麻烦帮忙回答一下，感激不尽🙏

**Reply by dxsup** [reply]
这里不讨论其他项目的细节，单独在kindling的discusssion里面开个问题吧，有时间我回复一下

**Comment by litaixun** [comment]
收到，感谢



---- 回复的原邮件 ----
| 发件人 | Daxin ***@***.***> |
| 日期 | 2025年02月25日 12:02 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [CloudDetail/apo] 欢迎大家积极提问🙋 (Discussion #4) |

这里不讨论其他项目的细节，单独在kindling的discusssion里面开个问题吧，有时间我回复一下
